File: app-miyou/src/main/java/com/miyou/app/infrastructure/credit/repository/CreditTransactionMongoRepository.java
package com.miyou.app.infrastructure.credit.repository;

import org.springframework.data.domain.Pageable;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;

import com.miyou.app.infrastructure.credit.document.CreditTransactionDocument;
import reactor.core.publisher.Flux;

public interface CreditTransactionMongoRepository
	extends
		ReactiveMongoRepository<CreditTransactionDocument, String> {

	Flux<CreditTransactionDocument> findByUserIdOrderByCreatedAtDesc(String userId,
		Pageable pageable);
}

File: app-miyou/src/main/kotlin/com/miyou/app/infrastructure/credit/repository/CreditTransactionMongoRepository.kt
package com.miyou.app.infrastructure.credit.repository

import com.miyou.app.infrastructure.credit.document.CreditTransactionDocument
import org.springframework.data.domain.Pageable
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Flux

interface CreditTransactionMongoRepository : ReactiveMongoRepository<CreditTransactionDocument, String> {
    fun findByUserIdOrderByCreatedAtDesc(
        userId: String,
        pageable: Pageable,
    ): Flux<CreditTransactionDocument>
}
